.changeset/good-camels-pull.md

```diff
@@ -0,0 +1,68 @@
+---
+'@astrojs/markdoc': minor
+'@astrojs/mdx': major
+'@astrojs/markdown-remark': major
+'astro': major
+---
+
+Updates Markdown heading ID generation
+
+In Astro 5.x, an additional default processing step to Markdown stripped trailing hyphens from the end of IDs for section headings ending in special characters. This provided a cleaner `id` value, but could lead to incompatibilities rendering your Markdown across platforms.
+
+In Astro 5.5, the `experimental.headingIdCompat` flag was introduced to allow you to make the IDs generated by Astro for Markdown headings compatible with common platforms like GitHub and npm, using the popular [`github-slugger`](https://github.com/Flet/github-slugger) package.
+
+Astro 6.0 removes this experimental flag and makes this the new default behavior in Astro: trailing hyphens from the end of IDs for headings ending in special characters are no longer removed.
+
+#### What should I do?
+
+If you have manual links to headings, you may need to update the anchor link value with a new trailing hyphen. For example, the following Markdown heading:
+
+```md
+## `<Picture />`
+```
+
+will now generate the following HTML with a trailing hyphen in the heading `id`:
+
+```html
+<h2 id="picture-"><code>&lt;Picture /&gt;</h2>
+```
+
+and must now be linked to as:
+
+```markdown
+See [the Picture component](/en/guides/images/#picture-) for more details.
+```
+
+If you were previously using this experimental feature, you must remove this experimental flag from your configuration as it no longer exists:
+
+```diff
+// astro.config.mjs
+import { defineConfig } from 'astro/config';
+
+export default defineConfig({
+  experimental: {
+-    headingIdCompat: true,
+  },
+})
+```
+
+If you were previously using the `rehypeHeadingIds` plugin directly to enforce compatibility, remove the `headingIdCompat` option as it no longer exists:
+
+```diff
+// astro.config.mjs
+import { defineConfig } from 'astro/config';
+import { rehypeHeadingIds } from '@astrojs/markdown-remark';
+import { otherPluginThatReliesOnHeadingIDs } from 'some/plugin/source';
+
+export default defineConfig({
+  markdown: {
+    rehypePlugins: [
+-     [rehypeHeadingIds, { headingIdCompat: true }],
++     [rehypeHeadingIds],
+      otherPluginThatReliesOnHeadingIDs,
+    ],
+  },
+});
+```
+
+If you want to keep the old ID generation for backward compatibility reasons, you can [create a custom rehype plugin that will generate headings IDs like Astro 5.x](https://docs.astro.build/en/guides/upgrade-to/v6/#changed-markdown-heading-id-generation) by following the instructions in the Astro 6 upgrade guide. This will allow you to continue to use your existing anchor links without adding trailing hyphens.
```

Comment:
```suggestion
If you have manual links to headings, you may need to update some anchor link values with a new trailing hyphen. 

If you were previously using this experimental feature, remove this experimental flag from your configuration.

If you were previously using the `rehypeHeadingIds` plugin directly to enforce compatibility, remove the `headingIdCompat` option as it no longer exists.

See the [Astro 6.0 upgrade guide](https://docs.astro.build/en/guides/upgrade-to/v6/#changed-markdown-heading-id-generation) for upgrade examples, and instructions to create a custom rehype plulgin if you want to keep the old ID generation for backward compatibility reasons.
```
So, if we're going to be making these shorter and linking to the guide, is this short enough?